lib/manageiq/password.rb

```diff
@@ -159,46 +116,54 @@ def self.load_v2_key
   passwords in your database.
   EOS
         Kernel.warn msg
+        nil
       end
     end
 
-    def self.add_legacy_key(filename, type = "alt")
-      key = load_key_file(filename, type != :v0)
-      keys[type.to_s] = key if key
-      key
+    def self.key=(key)
+      @@key = key
     end
 
-    # used by tests only
-    def self.v2_key=(key)
-      (@@all_keys ||= {})["v2"] = key
+    def self.add_legacy_key(filename, type = "alt")
```

Comment:
looking forward to the next pr that will drop this.
thanks for keeping it in for this pr, though.